WordPress/Classes/Services/PlanService.swift

```diff
@@ -241,15 +241,15 @@ extension PlanService {
 }
 
 private struct PlanStorage {
-    static func updateHasDomainCredit(_ planID: Int, forSite siteID: Int, hasDomainCredit: Bool, in context: NSManagedObjectContext) {
-        guard let blog = try? Blog.lookup(withID: siteID, in: context) else {
-            let error = "Tried to update a plan for a non-existing site (ID: \(siteID))"
+    static func updateHasDomainCredit(_ planID: Int, forBlog blog: Blog, hasDomainCredit: Bool, in context: NSManagedObjectContext) {
+        guard let blogInContext = try? context.existingObject(with: blog.objectID) as? Blog else {
```

Comment:
[Similar solution used in recent refactoring of Sharing Service](https://github.com/wordpress-mobile/WordPress-iOS/blob/157aee0f9d2e9429e50a863ca6d7ecefbafe5be9/WordPress/Classes/Services/SharingService.swift#L144)